server/src/test/java/org/opensearch/cluster/routing/WeightedRoutingServiceTests.java

```diff
@@ -362,4 +362,36 @@ public void onFailure(Exception e) {}
         weightedRoutingService.registerWeightedRoutingMetadata(request.request(), listener);
         assertTrue(countDownLatch.await(30, TimeUnit.SECONDS));
     }
+
+    public void testAddWeightedRoutingPassesWhenWeightOfDecommissionedAttributeStillZero() throws InterruptedException {
```

Comment:
do we need test for weight update failure case?

Reply:
These tests were already added as part of previous PR